crates/router/src/core/payments/helpers.rs

```diff
@@ -2592,7 +2592,7 @@ pub fn validate_customer_access(
                     .customer_id
```

Comment:
Just out of curiosity, why doesn't this simply do 
```rust
let is_not_same_customer = request.customer_id != payment_intent.customer_id;
```

Reply:
Hi @sahkal , what about the other suggestion?

```rust
let is_same_customer = request.customer_id == payment_intent.customer_id;
```
or with as_ref if needed.

Reply:
both request.customer_id and payment_intent.customer_id are optional field.